common/headtracker/types/client.go

```diff
@@ -15,4 +15,7 @@ type Client[H types.Head[BLOCK_HASH], S types.Subscription, ID types.ID, BLOCK_H
 	// SubscribeNewHead is the method in which the client receives new Head.
 	// It can be implemented differently for each chain i.e websocket, polling, etc
 	SubscribeNewHead(ctx context.Context, ch chan<- H) (S, error)
+	// LatestFinalizedBlock - returns the latest block that was marked as finalized. Only applicable for PoS chains
```

Comment:
nit: I think we shouldn't be specific here with the consensus algorithm and drop the PoS reference. For all we know there are chains out there with PoA or other variations that also implement finality tag support.